src/main/java/com/ait/lienzo/client/widget/panel/scrollbars/ScrollablePanelHandler.java

```diff
@@ -33,6 +33,9 @@
 
     static final int DEFAULT_INTERNAL_SCROLL_WIDTH  = 1;
 
+    int currentPanelWidth;
```

Comment:
I see you added two fields in the state for this class: `currentPanelWidth`, `currentPanelHeight`. I understand it's because of avoiding refreshes if the "size" was not changed. 

It looks good to me, but this means that whatever called to the `refresh` method in this handler class, who maybe expects the refresh stuff to be happening, it will not be executed now, as the handler itself now has this state to be able to compare and decide...

Sooo the Q is - should we really apply this decision in the handler or on whatever caller to it? Could it maybe make sense to call `refresh`, even when "size" didn't change, but to update any other stuff (eg: scrollbars), any internal state?  

Following me? WDYT? Just branistorming :)

Reply:
Hey @romartin, I understand your point. However, if you take a look at the context the refresh method is being called, we may not have all the variables to make decisions about refreshing the scrollbars size/position or not. In addition most of the calls to this method were linked to that mediator that was causing the components to run horizontally when dragging (we removed it). That's why I considered not delegating this logic and keeping it internal to the handler. 
WDYT?

Reply:
Hi @handreyrc, @romartin, I think it should be possible to check something like:
```java
void synchronizeScrollSize(final int width, final int height)
{
    if (getInternalScrollPanel().getElement().getStyle().getProperty("width") != Integer.toString(width)
        || getInternalScrollPanel().getElement().getStyle().getProperty("height") != Integer.toString(height)) {
            getInternalScrollPanel().setPixelSize(width, height);
            getPanel().fireLienzoPanelBoundsChangedEvent();
    }
}
```

I didn't test it at runtime, but I think it should work. If so, I think this approach suggested by Roger will simplify overall logic so it can be good to do.

 Just thinking aloud :) Thank you!

Reply:
@hasys Good point and It works like a charm!  I've already updated the code.
Yes sir, I learned a new trick today.